crates/goose/src/agents/agent.rs

```diff
@@ -1083,7 +1096,7 @@ impl Agent {
                                     }
                                 } else {
                                     let mut permission_manager = PermissionManager::default();
-                                    let (permission_check_result, enable_extension_request_ids) =
+                                    let (mut permission_check_result, enable_extension_request_ids) =
```

Comment:
the check_tool_permission below is of the same shape as the others, so should also be a tool inspector